include/deal.II/cgal/surface_mesh.h

```diff
@@ -0,0 +1,76 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2020 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii_cgal_surface_mesh_h
+#define dealii_cgal_surface_mesh_h
+
+#include <deal.II/base/config.h>
+
+#include <deal.II/fe/mapping.h>
+
+#include <deal.II/grid/tria.h>
+
+#include <deal.II/cgal/utilities.h>
+
+#ifdef DEAL_II_WITH_CGAL
+#  include <CGAL/Surface_mesh.h>
+
+DEAL_II_NAMESPACE_OPEN
+
+namespace CGALWrappers
+{
+  /**
+   * Build a CGAL::Surface_mesh from a deal.II cell.
+   *
+   * The class Surface_mesh implements a halfedge data structure and can be used
+   * to represent polyhedral surfaces. It is an edge-centered data structure
+   * capable of maintaining incidence information of vertices, edges, and faces.
+   * Each edge is represented by two halfedges with opposite orientation. The
+   * orientation of a face is chosen so that the halfedges around a face are
+   * oriented counterclockwise.
+   *
+   * More information on this class is available at
+   * https://doc.cgal.org/latest/Surface_mesh/index.html
+   *
+   * The function will throw an exception in dimension one. In dimension two it
+   * generates a surface mesh of the quadrilateral cell or of the triangle cell,
+   * while in dimension three it will generate the surface mesh of the cell,
+   * i.e., a polyhedral mesh containing the faces of the input cell.
+   *
+   * The generated mesh is useful when performing geometric operations using
+   * CGAL::Polygon_mesh_processing, i.e., to compute boolean operations on
+   * cells, splitting, cutting, slicing, etc.
+   *
+   * For examples on how to use the resulting  CGAL::Surface_mesh see
+   * https://doc.cgal.org/latest/Polygon_mesh_processing/
+   *
+   * @param[in] cell The input deal.II cell iterator
+   * @param[in] mapping The mapping used to map the vertices of the cell
+   * @param[out] mesh The output CGAL::Surface_mesh
+   */
+  template <typename CGALPointType, int dim, int spacedim>
+  void
+  to_cgal_mesh(
```

Comment:
I don't like the name of the function. I would prefer if the name would contain the information that the output is a surface mesh, e.g. `convert_to_cgal_surface_mesh()`.